packages/render-fidelity-tools/src/components/renderers/gltfpt-viewer.ts

```diff
@@ -0,0 +1,185 @@
+:host {
+  display: block;
+}
+`;
+  }
+
+  render() {
+    return html`<canvas id="canvas"></canvas>`;
+  }
+
+  private[$initialize]() {
+    this[$updateSize]();
+  }
+
+  private async[$updateScenario](scenario: ScenarioConfig) {
+    this[$canvas] = this.shadowRoot!.querySelector('canvas');
+
+    const dpr = resolveDpr();
+
+    const settings = {
+      pathtracing: true,
+      debugMode: 'None',
+      maxBounceDepth: 1,
+      useIBL: true,
+      autoScaleOnImport: false,
+      pixelRatio: 1.0 / dpr,  // TODO: clarify
+      autoRotate: false,
+      disableDirectShadows: true
+    };
+
+    this[$renderer] =
+        new PathtracingRenderer({'canvas': this[$canvas]!}, settings);
+
+    const renderer = this[$renderer];
+
+    await new Promise((resolve) => {
+      console.log('Loading resources for', scenario.model);
+      renderer.loadScene(scenario.model, scenario.lighting, () => {
+        this[$updateSize]();
+        // console.log(this[$canvas]!.width, this[$canvas]!.height);
+        renderer.resize(this[$canvas]!.width, this[$canvas]!.height)
+        resolve();
+      });
+    });
+
+    const numSamples = 8;
+
+    console.log('Rendering...');
+    renderer.render(
+        numSamples,
+        () => {
+          console.log('frame finished');
+        },
+        () => {
+          // Wait two rAFs to ensure we rendered at least once //TODO: clarify
+          requestAnimationFrame(() => {
+            requestAnimationFrame(() => {
+              this.dispatchEvent(new CustomEvent(
+                  'model-visibility', {detail: {visible: true}}));
+            });
+          });
+        });
+  }
+
+  private[$updateSize]() {
+    if (this[$canvas] == null || this.scenario == null) {
+      // Not initialized yet. This will be invoked again when initialized.
+      return;
+    }
+
+    // const Fov = Camera$Fov;
+    const canvas = this[$canvas]!;
+    const {dimensions, target, orbit, verticalFoV} = this.scenario;
+    console.log('Scenario:', dimensions, target, orbit, verticalFoV);
+
+    // const dpr = resolveDpr(); //TODO: clarify
+    const width = dimensions.width;    // * dpr;
+    const height = dimensions.height;  // * dpr;
+
+    canvas.width = width;
+    canvas.height = height;
+    canvas.style.width = `${dimensions.width}px`;
+    canvas.style.height = `${dimensions.height}px`;
+
+    const center = [target.x, target.y, target.z];
+
+    const theta = orbit.theta * Math.PI / 180;
+    const phi = orbit.phi * Math.PI / 180;
+    const radiusSinPhi = orbit.radius * Math.sin(phi);
+    const eye = [
+      radiusSinPhi * Math.sin(theta) + target.x,
+      orbit.radius * Math.cos(phi) + target.y,
+      radiusSinPhi * Math.cos(theta) + target.z
+    ];
+    if (orbit.radius <= 0) {
+      center[0] = eye[0] - Math.sin(phi) * Math.sin(theta);
+      center[1] = eye[1] - Math.cos(phi);
+      center[2] = eye[2] - Math.sin(phi) * Math.cos(theta);
+    }
+
+    const bbox = this[$renderer].getBoundingBox();
```

Comment:
This function is not found when we try to run locally. Is there a chance our dependency is not up to date?

Reply:
> @bsdorra When you update the dependencies in package.json, you also need to run `npm run update:package-lock`. This will allow it to start building properly. You should also pull master, as I updated out testing config for newer browser versions that helps out tests to pass.

Ok, I'll fix this. 

> This function is not found when we try to run locally. Is there a chance our dependency is not up to date?

The function was indeed missing. I pushed it yesterday when i was updating the config. It should work now.

Reply:
Thanks! Do you think you can push dspbr-pt up to npm? That would make tracking version updates like this more robust.